### ISSUE TITLE
Mac OS X backend: <Error>: CGContextClosePath: no current point 

### ISSUE THREAD

**Issue by lebigot** [matplotlib/matplotlib#3654]
With the Mac OS X backend, under certain conditions, I guess tons of error messages like:

``Oct 17 15:47:38 weinberg.local Python[877] <Error>: CGContextClosePath: no current point.``

when I move the mouse inside a slider *and* when the figure is first drawn (it contains many plots). The TkAgg backend does not report any problem.

This slows down considerably the interaction with the slider.

The code is currently complex, so sharing it here is not obvious: are there tests, changes, etc. that I could run so as to try to nail down the problem?

**Comment by mdehoon** [comment]
OK I get the same error message with your example.
Can you change line 265 in src/_macosx.m from
```C
            CGContextClosePath(cr);
```
to
```C
            if (n > 0) CGContextClosePath(cr);
```
and make sure the error message doesn't appear? It would be great if you could run the test suite to make sure this doesn't introduce new bugs.

**Comment by lebigot** [comment]
I never compiled Matplotlib (MacPorts!) and I will unfortunately not be able to do this right now…

**Comment by tacaswell** [comment]
@mdehoon Can you put in a PR with that patch?

**Comment by tacaswell** [comment]
Closed by #4006
